estate/models/estate_property.py

```diff
@@ -0,0 +1,123 @@
+from odoo import api, fields, models
+from odoo.exceptions import UserError, ValidationError
+from odoo.tools import float_compare, float_is_zero
+from datetime import datetime, timedelta
+
+class EstateProperty(models.Model):
+    _name = "estate.property"
+    _description = "Real Estate Property"
+    _order = "id desc"
+
+    # SQL Constraints
+    _sql_constraints = [
+        ('check_expected_price', 'CHECK(expected_price > 0)', 
+         'The expected price must be strictly positive.'),
+        ('check_selling_price', 'CHECK(selling_price >= 0)', 
+         'The selling price must be positive.'),
+    ]
+
+    name = fields.Char(required=True)
+    description = fields.Text()
+    postcode = fields.Char()
+    date_availability = fields.Date()
+    expected_price = fields.Float(required=True)
+    selling_price = fields.Float()
+    bedrooms = fields.Integer()
+    living_area = fields.Integer()
+    facades = fields.Integer()
+    garage = fields.Boolean()
+    garden = fields.Boolean()
+    garden_area = fields.Integer()
+    total_area = fields.Integer(compute="_compute_total_area", string="Total Area")
+    best_price = fields.Float(compute="_compute_best_price", string="Best Offer")
+    garden_orientation = fields.Selection([
+        ('north', 'North'),
+        ('south', 'South'),
+        ('east', 'East'),
+        ('west', 'West')
+    ])
+    active = fields.Boolean(default=True)
+    state = fields.Selection([
+        ('new', 'New'),
+        ('offer_received', 'Offer Received'),
+        ('offer_accepted', 'Offer Accepted'),
+        ('sold', 'Sold'),
+        ('cancelled', 'Cancelled')
+    ], required=True, copy=False, default='new')
+    property_type_id = fields.Many2one(
+        "estate.property.type",
+        string="Property Type"
+    )
+    buyer_id = fields.Many2one(
+        "res.partner",
+        string="Buyer",
+        copy=False
+    )
+    salesperson_id = fields.Many2one(
+        "res.users",
+        string="Salesperson",
+        default=lambda self: self.env.user
+    )
+    tag_ids = fields.Many2many(
+    "estate.property.tag",
+    string="Tags"
+    )
+    offer_ids = fields.One2many(
+    "estate.property.offer",
+    "property_id",
+    string="Offers"
+    )
+    
+    #api.depends area
+    @api.depends("living_area", "garden_area")
+    def _compute_total_area(self):
+        for record in self:
```

Comment:
```suggestion
        for property in self:
```
We tend to use a variable that show in which model we are so it is clearer for the person reading the code